text/0000-update-version-handling.md

```diff
@@ -0,0 +1,225 @@
+- Feature Name: update_version_handling
+- Start Date: 2019-04-23
+- RFC PR: (leave this empty)
+- Notion Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Change all version-specifying commands to use the `<tool>[@<version>]` syntax from `npm install` and `yarn add`.
+
+# Motivation
+[motivation]: #motivation
+
+Users may regularly want to add multiple tools to their toolchain at the same time. They are also accustomed to being able to install multiple dependencies simultaneously with the `npm install` or `yarn add` commands, both in a package and globally.
+
+- **Local installs:**
+
+    ```sh
+    npm install ember-cli@3.9.1 typescript@3
+    ```
+    
+    ```sh
+    yarn global ember-cli@3.9.1 typescript@3
+    ```
+
+- **Global installs:**
+
+    ```sh
+    npm install --global ember-cli@3.9.1 typescript@3
+    ```
+    
+    ```sh
+    yarn global add ember-cli@3.9.1 typescript@3
+    ```
+
+Notion currently does not support this, requiring individual invocations for each tool:
+
+- **User toolchain:**
+
+    ```sh
+    notion install ember-cli 3.9.1
+    notion install typescript 3
+    ```
+
+- **Project toolchain:**
+
+    ```sh
+    notion pin ember-cli 3.9.1
+    notion pin typescript 3
+    ```
+
+- **Tool retrieval:**
+
+    ```sh
+    notion fetch ember-cli 3.9.1
+    notion fetch typescript 3
+    ```
+
+This is a surprising difference for users coming from `npm` or `yarn`. Switching over to use the same invocation as the existing tools will make adoption smoother and less surprising.
+
+# Pedagogy
+[pedagogy]: #pedagogy
+
+Installation, fetching, and pinning function much as they do today, with two changes:
+
+- The tool + version combination is now a single string, in the format `tool[@<version>]`. That is, a user may supply a tool with its name only, e.g. `node`, or with a specific version using the `@` to separate the name from the version, e.g. `node@10`, `node@10.5`, `node@10.5.2`, or `node@latest`.
+
+- The invocations now allow multiple tool + version combinations, each of the format specified above. For example, a user may simultaneously pin a Node and a Yarn version (with any combination of the version specifier, or none at all):
+
+  ```sh
+  notion install node@10.5 yarn
+  ```
+
+## Explanation for Users
+
+### Current Node users
+
+There are two broad sets of current Node users: those not using a Node version management tool, and those using an existing tool like `nvm`, `nodenv`, or `nvm-windows`.
+
+Current Node users who are not using a Node version management tool should find this a simplification: it matches what npm and Yarn do.
```

Comment:
And also npx!

Reply:
Ah, yes I never got in the npx habit so I always forget it!